src/ctnetlink/message.rs

```diff
@@ -0,0 +1,209 @@
+// SPDX-License-Identifier: MIT
+
+use netlink_packet_utils::{
+    nla::DefaultNla, DecodeError, Emitable, Parseable, ParseableParametrized,
+};
+
+use crate::{
+    buffer::NetfilterBuffer,
+    constants::{
+        IPCTNL_MSG_CT_DELETE, IPCTNL_MSG_CT_GET, IPCTNL_MSG_CT_GET_CTRZERO,
+        IPCTNL_MSG_CT_GET_DYING, IPCTNL_MSG_CT_GET_STATS,
+        IPCTNL_MSG_CT_GET_STATS_CPU, IPCTNL_MSG_CT_GET_UNCONFIRMED,
+        IPCTNL_MSG_CT_NEW, NFNL_SUBSYS_CTNETLINK,
+    },
+};
+
+use super::nlas::{flow::nla::FlowNla, stat::nla::StatNla};
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub enum CtNetlinkMessage {
+    New(Vec<FlowNla>),
+    Get(Option<Vec<FlowNla>>),
+    Delete(Vec<FlowNla>),
+    GetCrtZero(Option<Vec<FlowNla>>),
+    GetStatsCPU(Option<Vec<StatNla>>),
+    GetStats(Option<Vec<StatNla>>),
+    GetDying(Option<Vec<FlowNla>>),
+    GetUnconfirmed(Option<Vec<FlowNla>>),
+    Other {
+        message_type: u8,
+        nlas: Vec<DefaultNla>,
```

Comment:
Unless kernel code confirmed future data will always a array of Nla, we should use `Other((u8, DefaultNla))`.

Reply:
It seems `NetfilterBuffer` doesn't provide the method to parse it to a single `DefaultNla`, should I create a new method?